rpc/server_auth.go

```diff
@@ -284,6 +284,11 @@ func (ss *simpleServer) tryAuth(ctx context.Context) (context.Context, error) {
 }
 
 func (ss *simpleServer) ensureAuthed(ctx context.Context) (context.Context, error) {
+	// Use handler if set (only used for testing).
```

Comment:
if this really should only be used during testing, should we add a guard around this? or maybe the comment should be `(should only be used for testing)`

Reply:
Extended the comment on `WithEnsureAuthedHandler` to say "should only be used for testing", and threw an error below in the case that we're in a testing environment `if !testing.Testing()`.

Reply:
[opt] we might want to consider either sticking this method into `testutils` or creating a separate test version of this package (something akin to https://pkg.go.dev/net/http/httptest). This prevents `testing` from being included in the final build of `rpc`.

That said, this might be annoying to do given that this method references a bunch of private methods, so I won't block on this.

Reply:
Oof yeah you're right we'll include `testing` in the build of `rpc`. How would you both feel about only having the comment "should only be used for testing" and no actual error from `ensureAuthed` within a testing environment?